### PR TITLE
Add Tensorflow Lite Micro as a module

### LINKED ISSUE #36947: Tensorflow: Dedicated tflite-micro repository
## Purpose

The tensorflow organisation has recently (June 25th) split out the tensorflow micro code into a dedicated repository, instead of being part of the parent tensorflow repo. The dedicated repository is considerably smaller (5.85MB instead of 680MB), and is where future development will take place. The old location of the tflite-micro code in the parent repository has been removed (https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/micro).

https://github.com/tensorflow/tensorflow
https://github.com/tensorflow/tflite-micro

We should swap over from the tensorflow fork (https://github.com/zephyrproject-rtos/tensorflow) to a fork of the tflite-micro repository.

### PR DESCRIPTION
**Pull request by laurenmurphyx64**
Adds TensorFlow Lite Micro as a module and makes some updates to reflect the change from using the larger TensorFlow repo to the new, smaller TensorFlow Lite Micro repo. Cannot be merged until [the module PR enabling Zephyr support](https://github.com/zephyrproject-rtos/tflite-micro/pull/1) is merged.

Fixes  #36947

### CODE REVIEW

**Comment by laurenmurphyx64** [comment]
(Note: The build will be broken until the PR for the module is merged.)

File: modules/tflite-micro/CMakeLists.txt
Comment (JordanYates) [inline modules/tflite-micro/CMakeLists.txt:125]: Why are all these headers part of `zephyr_library_sources`?

**Comment by nashif** [comment]
> (Note: The build and compliance checks will be broken until the PR for the module is merged.)

That does not need to be this way, you should actually be pointing to the PR in the west yaml file and not a specific commit. This is how we can verify that a module changes works with the Zephyr tree.

File: modules/tflite-micro/CMakeLists.txt
Comment (laurenmurphyx64) [reply modules/tflite-micro/CMakeLists.txt]: @JordanYates Removed in most recent commit. I put them in originally since they were listed in the TensorFlow-generated CMakeLists.txt next to  the source files. Sorry about that!

**Comment by laurenmurphyx64** [comment]
Recent updates:
- Fixing module name in tflite-micro repo (see [PR](https://github.com/zephyrproject-rtos/tflite-micro/pull/2)). Will change the Zephyr repo's west.yml from the PR revision to the zephyr branch revision once the PR is merged.
- Address feedback on headers included in library sources
- Update Hello World sources

File: modules/tflite-micro/CMakeLists.txt
Comment (JordanYates) [inline modules/tflite-micro/CMakeLists.txt:123]: Same deal with the LICENSE files, pretty sure they aren't needed to compile :)

Comment (laurenmurphyx64) [reply modules/tflite-micro/CMakeLists.txt]: Oops. That's a duh moment on my part. Removed!

**Review by JordanYates — APPROVED** [review]

File: samples/modules/tflite-micro/hello_world/src/main_functions.cc
Comment (chen-png) [inline samples/modules/tflite-micro/hello_world/src/main_functions.cc:122]: this indentation seems incorrect.

**Review by chen-png — APPROVED** [review]
LGTM

File: samples/modules/tflite-micro/hello_world/src/main_functions.cc
Comment (laurenmurphyx64) [reply samples/modules/tflite-micro/hello_world/src/main_functions.cc]: Good catch! Fixed in latest pushed commits.

**Comment by laurenmurphyx64** [comment]
**Note:** Please don't merge until https://github.com/zephyrproject-rtos/tflite-micro/pull/2 is merged.

**Comment by carlescufi** [comment]
> **Note:** Please don't merge until [zephyrproject-rtos/tflite-micro#2](https://github.com/zephyrproject-rtos/tflite-micro/pull/2) is merged.

It is merged now, can you update to the SHA?

**Pull request merged**
